### PR TITLE
Added quickstart sample for project running on Graal VM

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
@romain-grecourt please check pom.xm/Dockerfile, I would like to ask few questions about making it more user friendly

### CODE REVIEW

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+                            <execution>
+                                <id>default</id>
+                                <goals>
+                                    <goal>build</goal>
+                                    <!-- binds push goal to maven deploy -->
+                                    <!-- <goal>push</goal>-->
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: I would put the configuration under pluginManagement and add only executions in a profile.
That way someone can do `mvn package docker:build`.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                            <version>1.1.0</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: If the assumption is that the graalvm.home must be set, we should do something to enforce it.
Either a profile activated by the property or an execution to the enforcer plugin.
We need to make sure the the docker profile can be used when that property is not set.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        <dockerfile.plugin.version>1.4.10</dockerfile.plugin.version>
+        <main.class>io.helidon.examples.quickstart.graalvm.Main</main.class>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
+        <libs.classpath.prefix>libs</libs.classpath.prefix>
+        <copied.libs.dir>${project.build.directory}/${libs.classpath.prefix}</copied.libs.dir>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <argLine>-Dfile.encoding=UTF-8</argLine>
+        <native.resources>application.yaml|logging.properties</native.resources>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Let's not add a property for everything. I think this is fine to be hard-coded.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+    <groupId>io.helidon.examples</groupId>
+    <artifactId>quickstart-graalvm</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+    <name>${project.artifactId}</name>
+
+    <properties>
+        <!-- Graal VM home directory -->
+        <grallvm.home>/Library/Java/JavaVirtualMachines/graalvm-ce-${graalvm.version}/Contents/Home</grallvm.home>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: This should not be hard-coded. It needs to be in the user environment (either shell environment or settings.xml or command line property).

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+--><project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>io.helidon.examples</groupId>
+    <artifactId>quickstart-graalvm</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Make sure to update this to 1.0.X-SNAPSHOT (currently 1.0.3-SNAPSHOT)

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+                    <execution>
+                        <id>build-native-image</id>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                        <configuration>
+                            <executable>${grallvm.home}/bin/native-image</executable>
+                            <arguments>
+                                <argument>-H:Name=${basedir}/target/${native.name}</argument>
+                                <argument>-H:IncludeResources=${native.resources}</argument>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Does Graal support a config file for these instead of command line arguments ?

File: examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,25 @@
+FROM helidon/java-native:0.1.13 as build-native-image
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile]: Where is the source for `helidon-java-native:0.1.13` ? Did tomas put it in the workspace yet ?

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 5 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Set to 1.0.3-SNAPSHOT

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 3 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: native.resources removed and hard coded

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 4 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Graal VM is now expected as default JDK / JAVA_HOME

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 6 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: I should ask Graal VM guys about it. Keeping it as is now.

File: examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile (hunk 7 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile]: Probably yes because my default docker installation got this image without any problems

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 2 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Added enforcer plugin into native-image-docker and moved exec plugin to a separate profile activated by existence of ${java.home}/bin/native-image

**Comment by Tomas-Kraus** [comment]
This project was removed from archetype scripts. We can always put it back later.

File: examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile (hunk 7 above)
Comment (romain-grecourt) [reply examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile]: Dockerbuild does not resolve base images from source. Tomas did push this image to Dockerhub. The question is to prompt him to add the source to the workspace.

File: examples/quickstarts/archetypes/test-archetypes.sh [hunk 8]
```diff
@@ -69,6 +69,18 @@ mvn archetype:generate -DinteractiveMode=false \
 # build the generated project
 mvn -f ${PWD}/quickstart-mp/pom.xml install
 
+# invoke the quickstart-graalvm archetype
+#mvn archetype:generate -DinteractiveMode=false \
```
Comment (romain-grecourt) [inline examples/quickstarts/archetypes/test-archetypes.sh]: Can you remove these for now ?

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+        <dockerfile.plugin.version>1.4.10</dockerfile.plugin.version>
+        <main.class>io.helidon.examples.quickstart.graalvm.Main</main.class>
+        <maven.compiler.source>8</maven.compiler.source>
+        <maven.compiler.target>${maven.compiler.source}</maven.compiler.target>
+        <libs.classpath.prefix>libs</libs.classpath.prefix>
+        <copied.libs.dir>${project.build.directory}/${libs.classpath.prefix}</copied.libs.dir>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <argLine>-Dfile.encoding=UTF-8</argLine>
+        <main.port>8080</main.port>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: The port can be hard-coded too.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
+            <id>native-image-docker</id>
+            <build>
+                <pluginManagement>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Move pluginManagement in the default profile, no need to isolate these unless it conflicts with another profile.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+                                    <fail>true</fail>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <executions>
+                            <execution>
```
Comment (romain-grecourt) [inline examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Why is this needed for docker but not for native ?

File: examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile (hunk 7 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile]: He's on vacation so this won't be known this week.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 9 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: OK, done

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 11 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: native-image-exec profile works with 
```
<argument>-classpath</argument>
<classpath/>
```
but docker image needs all those files to be copied into it.

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 1 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: Common configuration block was moved to pluginManagement

File: examples/quickstarts/archetypes/test-archetypes.sh (hunk 8 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/archetypes/test-archetypes.sh]: OK, done

File: examples/quickstarts/helidon-quickstart-graalvm/pom.xml (hunk 10 above)
Comment (Tomas-Kraus) [reply examples/quickstarts/helidon-quickstart-graalvm/pom.xml]: OK, done.

File: examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile (hunk 7 above)
Comment (romain-grecourt) [reply examples/quickstarts/helidon-quickstart-graalvm/src/main/docker/Dockerfile]: ok.

**Comment by m0mus** [comment]
@romain-grecourt The original idea was to make an archetype. I would like to see archetype scripts as part of this PR. What's your objections?
